src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala

```diff
@@ -587,11 +596,14 @@ object SbtBundle extends AutoPlugin {
   private def formatServiceName(serviceName: String): String =
     s"""        service-name  = "$serviceName""""
 
-  private def formatEndpoints(endpoints: Map[String, Endpoint]): String = {
+  private def formatEndpoints(bundleConfVersion: BundleConfVersions.Value, endpoints: Map[String, Endpoint]): String = {
     val formatted =
       for {
         (label, Endpoint(bindProtocol, bindPort, services, serviceName, acls)) <- endpoints
       } yield {
+        if (acls.exists(_.nonEmpty) && bundleConfVersion != BundleConfVersions.V_1_2_0)
```

Comment:
Instead of `acls.exists(_.nonEmpty)` I'd check for `acls.isDefined`.

The current wouldn't throw an exception if the version is 1.1 and empty acls are specified: `Some(Set.empty)`.